hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestAvroSchemaResolutionSupport.scala

```diff
@@ -875,7 +875,7 @@ class TestAvroSchemaResolutionSupport extends HoodieClientTestBase with ScalaAss
 
     // after implicit type change, read the table with vectorized read enabled
     //fg reader with mor does not support vectorized currently and will auto read by row
-    if (HoodieSparkUtils.gteqSpark3_3 && !useFileGroupReader) {
+    if (HoodieSparkUtils.gteqSpark3_3 && (isCow || !useFileGroupReader)) {
```

Comment:
Should we run the query with vectorized reader in the else branch to make sure it works?
```
withSQLConf("spark.sql.parquet.enableNestedColumnVectorizedReader" -> "true") {
          readTable(tempRecordPath, useFileGroupReader)
        }
```